docs/devguide/contributing.rst

```diff
@@ -35,3 +35,86 @@ negative votes, the team will grant commit privileges.
 The core team will also be responsible for removing commit privileges when
 appropriate - for example for malicious merge behavior or just inactivity over
 an extended period of time.
+
+
+Quality gates for the pull requests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are three quality gates to ensure the pull requests quality, `Hound`_ for
+code style check, `Travis CI`_ for unit-test and coveralls, `Jenkins`_ for the combination 
+test including unit-test and smoke test.. When a pull request created, all tests 
+will run automatically, test results can be found in the merge status field of
+each pull request page.
+
+.. _Hound: https://houndci.com/
+.. _Travis CI: https://travis-ci.org/
+.. _Jenkins: https://jenkins.io/
+.. _jshint: http://jshint.com/
+
+**Hound**
+
+Hound works with `jshint`_ and comments on style violations in pull requests. 
+Configuration files ``.hound.yml`` and ``.jshintrc`` have been created in each 
+repository, so before creating a pull request, you can check code style locally with 
+jshint to find out style violations beforehand.
+
+**Travis CI**
+
+Travis CI runs the unit tests, and then does some potentially ancillary actions.
+The build specifics are detailed in the ``.travis.yml`` file within each repository.
+For finding out basic errors before creating a pull request, you can run unit test
+locally using ``npm test`` within each repository.
+
+**Jenkins**
+
+Jenkins uses the Github Pull Request Builder plugin to monitor all pull requests
+to perform quality gate tests prior to merge. The gates include running all the unit
+tests, running all dependent project unit tests with the code proposed from the pull 
+request, running an integration "smoke test" to verify basic end to end functionality
+and commenting on the details of test case failure. Jenkins is also able to takes 
+instructions from pull request comments to deal with complex test scenarios. 
+Instructions can be written in pull request description or comments.
+
+The following table show all the Jenkins Instructions and usage:
```

Comment:
What about mentioning use of a whitelist?  Folks not on the whitelist need to have a CC type in "ok to test" so their PRs can be tested prior to being merged.  This may change over time, but it is currently required.